airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -1202,6 +1202,22 @@ def create_concurrent_cursor_from_perpartition_cursor(
         )
         cursor_field = CursorField(interpolated_cursor_field.eval(config=config))
 
+        datetime_format = datetime_based_cursor_model.datetime_format
+
+        cursor_granularity = (
+            parse_duration(datetime_based_cursor_model.cursor_granularity)
+            if datetime_based_cursor_model.cursor_granularity
+            else None
+        )
+
+        connector_state_converter: DateTimeStreamStateConverter
+        connector_state_converter = CustomFormatConcurrentStreamStateConverter(
```

Comment:
nit: This is common logic with `create_concurrent_cursor_from_datetime_based_cursor`. Should we extract this in a method so that a change to one affects the other?